insonmnia/hub/acl.go

```diff
@@ -32,6 +35,34 @@ func NewACLStorage() ACLStorage {
 	}
 }
 
+func (s *workerACLStorage) MarshalJSON() ([]byte, error) {
+	if s == nil {
+		return json.Marshal(nil)
+	}
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	set := make([]string, 0)
+	s.storage.Each(func(item interface{}) bool {
+		set = append(set, item.(string))
+		return true
+	})
+	return json.Marshal(set)
+}
+
+func (s *workerACLStorage) UnmarshalJSON(data []byte) error {
+	unmarshalled := make([]string, 0)
+	err := json.Unmarshal(data, &unmarshalled)
+	if err != nil {
+		return err
+	}
+	s.storage = set.NewNonTS()
```

Comment:
/sarcasm what if we have more than one thread?

Reply:
What?